mongo/integration/change_stream_test.go

```diff
@@ -682,6 +684,43 @@ func TestChangeStream_ReplicaSet(t *testing.T) {
 		assert.True(mt, ok, "expected field 'allChangesForCluster' to be boolean, got %v", acfcVal.Type.String())
 		assert.False(mt, acfc, "expected field 'allChangesForCluster' to be false, got %v", acfc)
 	})
+
+	withBSONOpts := mtest.NewOptions().ClientOptions(
+		options.Client().SetBSONOptions(&options.BSONOptions{
+			UseJSONStructTags: true,
+		}))
+	mt.RunOpts("with BSONOptions", withBSONOpts, func(mt *mtest.T) {
+		cs, err := mt.Coll.Watch(context.Background(), mongo.Pipeline{})
+		require.NoError(mt, err, "Watch error")
+		defer closeStream(cs)
+
+		type myDocument struct {
+			A string `json:"x"`
```

Comment:
Why do we need a `json:"x"` tag?

Reply:
The test uses `BSONOptions` with `UseJSONStructTags: true`. To demonstrate that the `BSONOptions` configuration is used, we need to add a `json` struct tag that overrides the default field name.